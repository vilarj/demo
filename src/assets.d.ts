declare module '*.tiff' {
  const src: string;
  export default src;
}

declare module '*.ico' {
  const src: string;
  export default src;
}
