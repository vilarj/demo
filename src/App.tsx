import { useState, useMemo } from 'react';
import { Layout, Menu, Avatar, ConfigProvider } from 'antd';
import { HomeIcon, DocumentDuplicateIcon, UserIcon, BellIcon, Cog6ToothIcon } from '@heroicons/react/24/outline';
import { Routes, Route, useNavigate, useLocation } from 'react-router-dom';
import cumulusLogo from './assets/cumulus_logo.png';

import Home from './pages/Home';
import Demo from './pages/Demo';

const { Header, Content, Sider } = Layout;

const App = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const [collapsed, setCollapsed] = useState(false);

  const selectedKey = useMemo(() => {
    if (location.pathname === '/demo') return 'demo';
    return 'home';
  }, [location.pathname]);

  return (
    <ConfigProvider theme={{ token: { colorPrimary: '#00aff0', colorInfo: '#00aff0', fontFamily: 'Inter, "Source Sans 3", sans-serif' } }}>
      <Layout style={{ minHeight: '100vh' }}>
        <Sider collapsible collapsed={collapsed} onCollapse={setCollapsed}>
          <div className="flex items-center justify-center h-12 m-2">
            <img src={cumulusLogo} alt="Cumulus Logo" className="mt-5 h-8" />
          </div>
          <Menu
            theme="dark"
            mode="inline"
            selectedKeys={[selectedKey]}
            onClick={({ key }) => navigate(key === 'home' ? '/' : `/${key}`)}
            items={[
              {
                key: 'home',
                icon: <HomeIcon className="h-5 w-5" />,
                label: 'Home',
              },
              {
                key: 'demo',
                icon: <DocumentDuplicateIcon className="h-5 w-5" />,
                label: 'Demo',
              },
            ]}
          />
        </Sider>
        <Layout>
          <Header
            style={{
              paddingInline: 16,
              background: '#001628',
              display: 'flex',
              alignItems: 'center',
            }}
          >
            <div className="ml-auto flex gap-4 items-center">
              <BellIcon className="h-6 w-6 cursor-pointer text-white" />
              <Cog6ToothIcon className="h-6 w-6 cursor-pointer text-white" />
              <Avatar style={{ backgroundColor: 'transparent' }} icon={<UserIcon className="h-5 w-5 text-white" />} />
            </div>
          </Header>
          <Content style={{ margin: '16px' }}>
            <div className="p-6 bg-white rounded shadow-sm min-h-[360px]">
              <Routes>
                <Route path="/" element={<Home />} />
                <Route path="/demo" element={<Demo />} />
              </Routes>
            </div>
          </Content>
        </Layout>
      </Layout>
    </ConfigProvider>
  );
};

export default App;
