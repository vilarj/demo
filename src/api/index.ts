/**
 * Inventory Management System API
 *
 * This module provides a complete inventory management system with both a unified API
 * and individual services for managing tools, employees, assignments, and certificates.
 */

// Export the main InventoryAPI (recommended for most use cases)
export { InventoryAPI } from './services/InventoryAPI';

// Export individual services for granular control
export { AssignmentService } from './services/AssignmentService';
export { CertificateService } from './services/CertificateService';
export { EmployeeService } from './services/EmployeeService';
export { ToolService } from './services/ToolService';

// Export all types for external consumption
export type { IPaginatedResponse } from './interfaces/IPaginatedResponse';
export type { IPaginationOptions } from './interfaces/IPaginationOptions';
export type { AssigmentResult, ISO8601Date, UnassignResult } from './types/AssignmentType';
export type { Employee, EmployeeDirectory, EmployeeId } from './types/EmployeeInventoryType';
export type { Tool, ToolId, ToolInventory, ToolType } from './types/ToolInventoryType';
