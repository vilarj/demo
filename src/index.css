@import url('https://fonts.googleapis.com/css2?family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&family=Source+Sans+3:ital,wght@0,200..900;1,200..900&display=swap');
@import 'tailwindcss';

html,
body,
#root {
  height: 100%;
  font-family: 'Inter', 'Source Sans 3', sans-serif;
}

body {
  margin: 0;
}

.ant-menu-dark {
  margin: 30px 0 0 0;
  padding: 0 10px;
}

/* Responsive fixes for Actions column */
@media (max-width: 768px) {
  .inventory-table .ant-table-thead > tr > th:last-child,
  .inventory-table .ant-table-tbody > tr > td:last-child {
    min-width: 160px !important;
  }

  .inventory-table .ant-table-tbody > tr > td:last-child .ant-btn {
    font-size: 11px !important;
    padding: 0 2px !important;
  }

  /* Employee selector responsive adjustments */
  .inventory-table .ant-select {
    max-width: 200px !important;
    min-width: 150px !important;
  }

  .inventory-table .ant-select .ant-select-selector {
    padding: 0 8px !important;
  }
}

@media (max-width: 480px) {
  .inventory-table .ant-table-thead > tr > th:last-child,
  .inventory-table .ant-table-tbody > tr > td:last-child {
    min-width: 140px !important;
  }

  .inventory-table .ant-table-tbody > tr > td:last-child .ant-space-item {
    font-size: 10px !important;
  }

  .inventory-table .ant-table-tbody > tr > td:last-child .ant-btn {
    font-size: 10px !important;
    padding: 0 1px !important;
    min-width: 45px !important;
  }

  /* Employee selector for mobile screens */
  .inventory-table .ant-select {
    max-width: 140px !important;
    min-width: 120px !important;
    font-size: 12px !important;
  }

  .inventory-table .ant-select .ant-select-selector {
    padding: 0 6px !important;
    font-size: 12px !important;
  }

  .inventory-table .ant-select .ant-select-selection-placeholder {
    font-size: 11px !important;
  }
}
