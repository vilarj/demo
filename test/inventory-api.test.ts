import type { EmployeeId, ISO8601Date, ToolId } from '../src/inventory-api';
import { MockInventorySystem } from '../src/inventory-api';

const today = (): ISO8601Date => new Date().toISOString().split('T')[0] as ISO8601Date;

describe('MockInventorySystem', () => {
  beforeAll(() => {
    jest.spyOn(Date.prototype, 'toISOString').mockReturnValue('2025-06-08T00:00:00.000Z');
  });
  afterAll(() => {
    jest.restoreAllMocks();
  });

  describe('assign', () => {
    it('assignTool assigns tool if unassigned and not overdue', async () => {
      expect.hasAssertions();
      const employeeId: EmployeeId = 'E1';
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: null,
            assignedOn: null,
          },
        },
        { E1: { id: employeeId, name: 'Alice' } },
        0,
      );
      const result = await system.assignTool({ toolId, employeeId });
      const tool = await system.getTool(toolId);
      expect({ result, tool }).toMatchSnapshot();
    });

    it('assignTool invalid if already assigned to same employee', async () => {
      expect.hasAssertions();
      const employeeId: EmployeeId = 'E1';
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: 'E1',
            assignedOn: today(),
          },
        },
        { E1: { id: employeeId, name: 'Alice' } },
        0,
      );
      const result = await system.assignTool({ toolId, employeeId });
      const tool = await system.getTool(toolId);
      expect({ result, tool }).toMatchSnapshot();
    });

    it('assignTool invalid already assigned to another employee', async () => {
      expect.hasAssertions();
      const employeeId: EmployeeId = 'E1';
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: 'E2' as EmployeeId,
            assignedOn: today(),
          },
        },
        {
          E1: { id: employeeId, name: 'Alice' },
          E2: { id: 'E2', name: 'Bob' },
        },
        0,
      );
      const result = await system.assignTool({ toolId, employeeId });
      const tool = await system.getTool(toolId);
      expect({ result, tool }).toMatchSnapshot();
    });

    it('assignTool invalid if tool is overdue for calibration', async () => {
      expect.hasAssertions();
      const employeeId: EmployeeId = 'E1';
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: '2000-01-01' as ISO8601Date,
            assignedTo: null,
            assignedOn: null,
          },
        },
        { E1: { id: employeeId, name: 'Alice' } },
        0,
      );
      const result = await system.assignTool({ toolId, employeeId });
      const tool = await system.getTool(toolId);
      expect({ result, tool }).toMatchSnapshot();
    });
  });

  describe('reassign', () => {
    it('reassignTool reassigns if already assigned', async () => {
      expect.hasAssertions();
      const employeeId1: EmployeeId = 'E1';
      const employeeId2: EmployeeId = 'E2';
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: employeeId2,
            assignedOn: today(),
          },
        },
        {
          E1: { id: employeeId1, name: 'Alice' },
          E2: { id: employeeId2, name: 'Bob' },
        },
        0,
      );
      const result = await system.reassignTool({
        toolId,
        employeeId: employeeId1,
      });
      const tool = await system.getTool(toolId);
      expect({ result, tool }).toMatchSnapshot();
    });

    it('reassignTool invalid if not assigned', async () => {
      expect.hasAssertions();
      const employeeId: EmployeeId = 'E1';
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: null,
            assignedOn: null,
          },
        },
        { E1: { id: employeeId, name: 'Alice' } },
        0,
      );
      const result = await system.reassignTool({
        toolId,
        employeeId,
      });
      const tool = await system.getTool(toolId);
      expect({ result, tool }).toMatchSnapshot();
    });
  });

  describe('getTool', () => {
    it('returns a deep clone of the tool if it exists', async () => {
      expect.hasAssertions();
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: null,
            assignedOn: null,
          },
        },
        {},
        0,
      );
      const result = await system.getTool(toolId);
      const again = await system.getTool(toolId);
      expect({ result }).toMatchSnapshot();
      // Mutate result and check original is not affected (deep clone)
      if (result) result.model = 'CHANGED';
      expect(again?.model).not.toBe('CHANGED');
    });

    it('returns null if the tool does not exist', async () => {
      expect.hasAssertions();
      const system = new MockInventorySystem({}, {}, 0);
      const result = await system.getTool('T999' as ToolId);
      expect({ result }).toMatchSnapshot();
    });
  });

  describe('getEmployee', () => {
    it('returns a deep clone of the employee if it exists', async () => {
      expect.hasAssertions();
      const employeeId: EmployeeId = 'E1';
      const system = new MockInventorySystem(
        {},
        {
          E1: { id: employeeId, name: 'Alice' },
        },
        0,
      );
      const result = await system.getEmployee(employeeId);
      expect(result).toBeTruthy();
      expect({ result }).toMatchSnapshot();
      // Mutate result and check original is not affected (deep clone)
      result!.name = 'CHANGED';
      const again = await system.getEmployee(employeeId);
      expect(again?.name).not.toBe('CHANGED');
    });

    it('returns null if the employee does not exist', async () => {
      expect.hasAssertions();
      const system = new MockInventorySystem({}, {}, 0);
      const result = await system.getEmployee('E999' as EmployeeId);
      expect({ result }).toMatchSnapshot();
    });
  });

  describe('getTools', () => {
    it('returns a deep clone of all tools', async () => {
      expect.hasAssertions();
      const toolId: ToolId = 'T1';
      const system = new MockInventorySystem(
        {
          T1: {
            id: toolId,
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: null,
            assignedOn: null,
          },
        },
        {},
        0,
      );
      const result = await system.getTools();
      expect({ result }).toMatchSnapshot();
      // Mutate result and check original is not affected (deep clone)
      result[0]!.model = 'CHANGED';
      const again = await system.getTools();
      expect(again[0]?.model).not.toBe('CHANGED');
    });

    it('returns an empty array if there are no tools', async () => {
      expect.hasAssertions();
      const system = new MockInventorySystem({}, {}, 0);
      const result = await system.getTools();
      expect({ result }).toMatchSnapshot();
    });
  });

  describe('getTools (filtering)', () => {
    let system: MockInventorySystem;
    beforeAll(() => {
      system = new MockInventorySystem(
        {
          T1: {
            id: 'T1',
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: null,
            assignedOn: null,
          },
          T2: {
            id: 'T2',
            type: 'HydraulicWrench',
            model: 'HW-1',
            serialNumber: 'SN123',
            calibrationDueDate: today(),
            assignedTo: 'E1',
            assignedOn: '2025-06-01',
          },
        },
        {
          E1: { id: 'E1', name: 'Alice' },
        },
        0,
      );
    });
    it('return all tools by default', async () => {
      expect.hasAssertions();
      const result = await system.getTools();
      expect({ result }).toMatchSnapshot();
    });
    it('all filter', async () => {
      expect.hasAssertions();
      const result = await system.getTools('all');
      expect({ result }).toMatchSnapshot();
    });
    it('assigned filter', async () => {
      expect.hasAssertions();
      const result = await system.getTools('assigned');
      expect({ result }).toMatchSnapshot();
    });
    it('available filter', async () => {
      expect.hasAssertions();
      const result = await system.getTools('available');
      expect({ result }).toMatchSnapshot();
    });
  });

  describe('getEmployees', () => {
    it('returns a deep clone of all employees', async () => {
      expect.hasAssertions();
      const employeeId: EmployeeId = 'E1';
      const system = new MockInventorySystem(
        {},
        {
          E1: { id: employeeId, name: 'Alice' },
        },
        0,
      );
      const result = await system.getEmployees();
      expect({ result }).toMatchSnapshot();
      result[0]!.name = 'CHANGED';
      const again = await system.getEmployees();
      expect(again[0]?.name).not.toBe('CHANGED');
    });

    it('returns an empty array if there are no employees', async () => {
      expect.hasAssertions();
      const system = new MockInventorySystem({}, {}, 0);
      const result = await system.getEmployees();
      expect({ result }).toMatchSnapshot();
    });
  });
});
