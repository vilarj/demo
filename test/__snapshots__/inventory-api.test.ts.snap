// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryAPI assign assignTool assigns tool if unassigned and not overdue 1`] = `
{
  "result": {
    "ok": true,
    "tool": {
      "assignedOn": "2025-06-08",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  },
  "tool": {
    "assignedOn": null,
    "assignedTo": null,
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`InventoryAPI assign assignTool invalid already assigned to another employee 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is already assigned to another employee: E2.",
    "ok": false,
  },
  "tool": {
    "assignedOn": "2025-06-08",
    "assignedTo": "E2",
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`InventoryAPI assign assignTool invalid if already assigned to same employee 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is already assigned to employee E1.",
    "ok": false,
  },
  "tool": {
    "assignedOn": "2025-06-08",
    "assignedTo": "E1",
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`InventoryAPI assign assignTool invalid if tool is overdue for calibration 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is overdue for calibration and cannot be assigned.",
    "ok": false,
  },
  "tool": {
    "assignedOn": null,
    "assignedTo": null,
    "calibrationDueDate": "2000-01-01",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`InventoryAPI getEmployee returns a deep clone of the employee if it exists 1`] = `
{
  "result": {
    "id": "E1",
    "name": "Alice",
  },
}
`;

exports[`InventoryAPI getEmployee returns null if the employee does not exist 1`] = `
{
  "result": null,
}
`;

exports[`InventoryAPI getEmployees returns a deep clone of all employees 1`] = `
{
  "result": [
    {
      "id": "E1",
      "name": "Alice",
    },
  ],
}
`;

exports[`InventoryAPI getEmployees returns an empty array if there are no employees 1`] = `
{
  "result": [],
}
`;

exports[`InventoryAPI getTool returns a deep clone of the tool if it exists 1`] = `
{
  "result": {
    "assignedOn": null,
    "assignedTo": null,
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`InventoryAPI getTool returns null if the tool does not exist 1`] = `
{
  "result": null,
}
`;

exports[`InventoryAPI getTools (filtering) all filter 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
    {
      "assignedOn": "2025-06-01",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T2",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`InventoryAPI getTools (filtering) assigned filter 1`] = `
{
  "result": [
    {
      "assignedOn": "2025-06-01",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T2",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`InventoryAPI getTools (filtering) available filter 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`InventoryAPI getTools (filtering) return all tools by default 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
    {
      "assignedOn": "2025-06-01",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T2",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`InventoryAPI getTools returns a deep clone of all tools 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`InventoryAPI getTools returns an empty array if there are no tools 1`] = `
{
  "result": [],
}
`;

exports[`InventoryAPI reassign reassignTool invalid if not assigned 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is not currently assigned to any employee.",
    "ok": false,
  },
  "tool": {
    "assignedOn": null,
    "assignedTo": null,
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`InventoryAPI reassign reassignTool reassigns if already assigned 1`] = `
{
  "result": {
    "ok": true,
    "tool": {
      "assignedOn": "2025-06-08",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  },
  "tool": {
    "assignedOn": "2025-06-08",
    "assignedTo": "E2",
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`MockInventorySystem assign assignTool assigns tool if unassigned and not overdue 1`] = `
{
  "result": {
    "ok": true,
    "tool": {
      "assignedOn": "2025-06-08",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  },
  "tool": {
    "assignedOn": "2025-06-08",
    "assignedTo": "E1",
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`MockInventorySystem assign assignTool invalid already assigned to another employee 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is already assigned to another employee: E2.",
    "ok": false,
  },
  "tool": {
    "assignedOn": "2025-06-08",
    "assignedTo": "E2",
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`MockInventorySystem assign assignTool invalid if already assigned to same employee 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is already assigned to employee E1.",
    "ok": false,
  },
  "tool": {
    "assignedOn": "2025-06-08",
    "assignedTo": "E1",
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`MockInventorySystem assign assignTool invalid if tool is overdue for calibration 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is overdue for calibration and cannot be assigned.",
    "ok": false,
  },
  "tool": {
    "assignedOn": null,
    "assignedTo": null,
    "calibrationDueDate": "2000-01-01",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`MockInventorySystem getEmployee returns a deep clone of the employee if it exists 1`] = `
{
  "result": {
    "id": "E1",
    "name": "Alice",
  },
}
`;

exports[`MockInventorySystem getEmployee returns null if the employee does not exist 1`] = `
{
  "result": null,
}
`;

exports[`MockInventorySystem getEmployees returns a deep clone of all employees 1`] = `
{
  "result": [
    {
      "id": "E1",
      "name": "Alice",
    },
  ],
}
`;

exports[`MockInventorySystem getEmployees returns an empty array if there are no employees 1`] = `
{
  "result": [],
}
`;

exports[`MockInventorySystem getTool returns a deep clone of the tool if it exists 1`] = `
{
  "result": {
    "assignedOn": null,
    "assignedTo": null,
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`MockInventorySystem getTool returns null if the tool does not exist 1`] = `
{
  "result": null,
}
`;

exports[`MockInventorySystem getTools (filtering) all filter 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
    {
      "assignedOn": "2025-06-01",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T2",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`MockInventorySystem getTools (filtering) assigned filter 1`] = `
{
  "result": [
    {
      "assignedOn": "2025-06-01",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T2",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`MockInventorySystem getTools (filtering) available filter 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`MockInventorySystem getTools (filtering) return all tools by default 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
    {
      "assignedOn": "2025-06-01",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T2",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`MockInventorySystem getTools returns a deep clone of all tools 1`] = `
{
  "result": [
    {
      "assignedOn": null,
      "assignedTo": null,
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  ],
}
`;

exports[`MockInventorySystem getTools returns an empty array if there are no tools 1`] = `
{
  "result": [],
}
`;

exports[`MockInventorySystem reassign reassignTool invalid if not assigned 1`] = `
{
  "result": {
    "error": "Tool with ID T1 is not currently assigned to any employee.",
    "ok": false,
  },
  "tool": {
    "assignedOn": null,
    "assignedTo": null,
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;

exports[`MockInventorySystem reassign reassignTool reassigns if already assigned 1`] = `
{
  "result": {
    "ok": true,
    "tool": {
      "assignedOn": "2025-06-08",
      "assignedTo": "E1",
      "calibrationDueDate": "2025-06-08",
      "id": "T1",
      "model": "HW-1",
      "serialNumber": "SN123",
      "type": "HydraulicWrench",
    },
  },
  "tool": {
    "assignedOn": "2025-06-08",
    "assignedTo": "E1",
    "calibrationDueDate": "2025-06-08",
    "id": "T1",
    "model": "HW-1",
    "serialNumber": "SN123",
    "type": "HydraulicWrench",
  },
}
`;
